### PR TITLE
- added logger control functions to new api

### PR DESCRIPTION
**Pull request by aschnell**

### CODE REVIEW

File: storage/Logger.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+namespace storage
+{
+
+    class Logger
+    {
+    public:
+
+	virtual ~Logger() {}
+
+	virtual bool test(int level);
```
Comment (jreidinger) [inline storage/Logger.h]: what is purpose of this method? it always return true.

**Comment by jreidinger** [comment]
in general I think it really helps if new API for logger have documentation stating what is its intention and intention of its methods.

File: storage/Logger.cc [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#include <iostream>
+
+#include "storage/Logger.h"
+#include "storage/Utils/AppUtil.h"
+
+
+namespace storage
+{
+
+    Logger* my_logger = nullptr;
```
Comment (shundhammer) [inline storage/Logger.cc]: Add 'static' ? This does not need to be exposed to the outside world.

File: storage/Logger.h [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+{
+
+    /**
+     * The Logger class.
+     */
+    class Logger
+    {
+    public:
+
+	virtual ~Logger() {}
```
Comment (shundhammer) [inline storage/Logger.h]: Even if it's just as empty as the dtor, a simple ctor would be nice.

File: storage/Utils/AppUtil.cc [hunk 4]
```diff
@@ -368,13 +374,33 @@ bool isNfsDev( const string& dev )
...
+				  content.substr(pos1, pos2 - pos1));
 		if (pos2 == string::npos)
 		    break;
 		pos1 = pos2 + 1;
 	    }
 	}
+	else
+	{
+	    // legacy
+	    CallbackLogDo pfc = getLogDoCallback();
```
Comment (shundhammer) [inline storage/Utils/AppUtil.cc]: pfc? Private First Class? ;-)

Maybe use something more speaking like, say, app_logger_callback() ? it's only used twice in the function, and that name change would make quite some difference when trying to understand the code.

File: storage/Logger.cc (hunk 2 above)
Comment (aschnell) [inline storage/Logger.cc]: OK.

File: storage/Utils/AppUtil.cc (hunk 4 above)
Comment (aschnell) [inline storage/Utils/AppUtil.cc]: That is old code which will be removed when we can drop the compatibility layer.

**Comment by shundhammer** [comment]
LGTM

**Pull request merged**